File: src/main/java/entities/gestionevents/Black_Listed.java
package entities.gestionevents;

import java.sql.Date;

public class Black_Listed {
    private int id_user;
    private Date start_ban;
    private Date end_ban;

    public Black_Listed(int id_user, Date start_ban, Date end_ban) {
        this.id_user = id_user;
        this.start_ban = start_ban;
        this.end_ban = end_ban;
    }
    public Black_Listed()
    {}
    public int getId_user() {
        return id_user;
    }
    public Date getStart_ban() {
        return start_ban;
    }
    public Date getEnd_ban() {
        return end_ban;
    }
    public void setId_user(int id_user) {
        this.id_user = id_user;
    }
    public void setStart_ban(Date start_ban) {
        this.start_ban = start_ban;
    }
    public void setEnd_ban(Date end_ban) {
        this.end_ban = end_ban;
    }



}

File: webapp/src/GymPlus/src/Entity/BlackListed.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\BlackListedRepository;

#[ORM\Entity (repositoryClass: BlackListedRepository::class)]
class BlackListed
{
    #[ORM\Column (options: ["default" => "CURRENT_TIMESTAMP"])]
    private ?\DateTime $startBan;

    #[ORM\Column (options: ["default" => "CURRENT_TIMESTAMP"])]
    private ?\DateTime $endBan;

    #[ORM\Id]
    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(nullable: false)]
    private $idUser;

    public function getStartBan(): ?\DateTimeInterface
    {
        return $this->startBan;
    }

    public function setStartBan(\DateTimeInterface $startBan): static
    {
        $this->startBan = $startBan;

        return $this;
    }

    public function getEndBan(): ?\DateTimeInterface
    {
        return $this->endBan;
    }

    public function setEndBan(\DateTimeInterface $endBan): static
    {
        $this->endBan = $endBan;

        return $this;
    }

    public function getIdUser(): ?User
    {
        return $this->idUser;
    }

    public function setIdUser(?User $idUser): static
    {
        $this->idUser = $idUser;

        return $this;
    }


}
